x-pack/plugins/lens/public/xy_visualization/xy_suggestions.ts

```diff
@@ -71,7 +71,9 @@ export function getSuggestions({
 
   if (
     (incompleteTable && state && !subVisualizationId) ||
-    table.columns.some((col) => col.operation.isStaticValue)
+    table.columns.some((col) => col.operation.isStaticValue) ||
+    // do not use suggestions with non-numeric metrics
+    table.columns.some((col) => !col.operation.isBucketed && col.operation.dataType !== 'number')
```

Comment:
nit: can we merge it with previous `table.columns.some`.  Not sure that we need 2 iterations by table.columns array